src/test/java/com/google/cloud/spanner/pgadapter/ProtocolTest.java

```diff
@@ -1209,7 +1210,75 @@ public void testCopyDataMessage() throws Exception {
     CopyDataMessage messageSpy = (CopyDataMessage) Mockito.spy(message);
     messageSpy.send();
 
-    Mockito.verify(mb, Mockito.times(1)).buildMutation(connectionHandler, payload);
+    Mockito.verify(mw, Mockito.times(1)).addMutations(connectionHandler, payload);
+  }
+
+  @Test
+  public void testMultipleCopyDataMessages() throws Exception {
+    Mockito.when(connection.createStatement()).thenReturn(statement);
+    Mockito.when(connection.prepareStatement(ArgumentMatchers.anyString()))
+        .thenReturn(preparedStatement);
+    Mockito.when(connectionHandler.getJdbcConnection()).thenReturn(connection);
+    Mockito.when(statement.getUpdateCount()).thenReturn(1);
+
+    byte[] messageMetadata = {'d'};
```

Comment:
this test is sending multiple copy data messages, but each message has a full row of data, which was already supported before this PR.

Can you modify the test to split a single row over multiple CopyData messages. For example, the first payload will be "1\t'o" and the second payload will be "ne'\n"

Reply:
The tests splits the data across rows now.

Reply:
looks good. thanks!